### PR TITLE
WC-1473: Add support for script content and script settings read/update endpoints

### PR DESCRIPTION
**Pull request by echen67**
This PR is to support new endpoints that allow the user to read/update either only the script content or only the script settings of a worker, as opposed to currently having to always update everything together. This makes it easier on the user when they only need to deal with one aspect of their worker.

## Has your change been tested?

I have added new tests to cover this change.

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [x] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

File: workers.go [hunk 1]
```diff
@@ -297,6 +303,139 @@ func (api *API) UploadWorker(ctx context.Context, rc *ResourceContainer, params
 	return r, nil
 }
 
+// GetScriptContent returns the pure script content of a worker.
+//
+// API reference: TODO:
+func (api *API) GetScriptContent(ctx context.Context, rc *ResourceContainer, scriptName string) (string, error) {
```
Comment (jacobbednarz) [inline workers.go]:
```suggestion
func (api *API) GetWorkersScriptContent(ctx context.Context, rc *ResourceContainer, scriptName string) (string, error) {
```

File: workers.go [hunk 2]
```diff
@@ -297,6 +303,139 @@ func (api *API) UploadWorker(ctx context.Context, rc *ResourceContainer, params
...
+		return "", err
+	}
+
+	return string(res.Body), nil
+}
+
+// PutScriptContent pushes only script content, no metadata.
+//
+// API reference: TODO:
+func (api *API) PutScriptContent(ctx context.Context, rc *ResourceContainer, params CreateWorkerParams) (WorkerScriptResponse, error) {
```
Comment (jacobbednarz) [inline workers.go]:
```suggestion
func (api *API) UpdateWorkersScriptContent(ctx context.Context, rc *ResourceContainer, params UpdateWorkersScriptContentParams) (WorkerScriptResponse, error) {
```

(and need to define `UpdateWorkersScriptContentParams`)

File: workers.go [hunk 3]
```diff
@@ -297,6 +303,139 @@ func (api *API) UploadWorker(ctx context.Context, rc *ResourceContainer, params
...
+		return r, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return r, nil
+}
+
+// GetScriptSettings returns the metadata of a worker.
+//
+// API reference: TODO:
+func (api *API) GetScriptSettings(ctx context.Context, rc *ResourceContainer, scriptName string) (WorkerScriptSettingsResponse, error) {
```
Comment (jacobbednarz) [inline workers.go]:
```suggestion
func (api *API) GetWorkersScriptSettings(ctx context.Context, rc *ResourceContainer, scriptName string) (WorkerScriptSettingsResponse, error) {
```

File: workers.go [hunk 4]
```diff
@@ -297,6 +303,139 @@ func (api *API) UploadWorker(ctx context.Context, rc *ResourceContainer, params
...
+
+	r.Success = true
+
+	return r, nil
+}
+
+// PatchScriptSettings pushes only script metadata.
+//
+// API reference: TODO:
+func (api *API) PatchScriptSettings(ctx context.Context, rc *ResourceContainer, params CreateWorkerParams) (WorkerScriptSettingsResponse, error) {
```
Comment (jacobbednarz) [inline workers.go]:
```suggestion
func (api *API) UpdateWorkersScriptSettings(ctx context.Context, rc *ResourceContainer, params UpdateWorkersScriptSettingsParams) (WorkerScriptSettingsResponse, error) {
```

(need to define `UpdateWorkersScriptSettingsParams`)

**Comment by jacobbednarz** [comment]
looks good in general, just a couple of notes from https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#methods that we should update here for consistency.

**Comment by jacobbednarz** [comment]
thanks @echen67, this looks awesome. once we have those API schemas land, we're good to go here.

File: workers.go [hunk 5]
```diff
@@ -297,6 +346,144 @@ func (api *API) UploadWorker(ctx context.Context, rc *ResourceContainer, params
...
+func (api *API) GetWorkersScriptContent(ctx context.Context, rc *ResourceContainer, scriptName string) (string, error) {
+	if rc.Level != AccountRouteLevel {
+		return "", ErrRequiredAccountLevelResourceContainer
+	}
+
+	if rc.Identifier == "" {
+		return "", ErrMissingAccountID
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/workers/scripts/%s/content/v2", rc.Identifier, scriptName)
```
Comment (jacobbednarz) [inline workers.go:361]: the [API docs](https://developers.cloudflare.com/api/operations/worker-script-put-content) suggest this is just `/accounts/:account_id/workers/scripts/:script/content`. is that correct?

Comment (echen67) [reply workers.go:361]: Ah, technically both work, but /v2 is preferred for this GET endpoint so I'll update the API docs to match. Thanks for catching this!

**Comment by jacobbednarz** [comment]
thanks for getting this one over the line @echen67 🚀

**Pull request merged**